addons/portal/views/portal_templates.xml

```diff
@@ -550,7 +550,72 @@
+                            <div class="modal-header">
+                                <h5 class="modal-title">Deactivate Account</h5>
+                                <button type="button" class="close" data-dismiss="modal">
+                                    ×
+                                </button>
+                            </div>
+                            <form action="/my/deactivate_account" method="post" class="modal-body"
+                                id="portal_deactivate_account_form">
+                                <div>
+                                    <div class="alert alert-danger"
+                                        t-esc="get_error(errors, 'deactivate.other')"/>
+                                    <p class="text-muted">
+                                        Disable your account, preventing any further login.<br/>
+                                        <b>This action cannot be undone.</b>
+                                    </p>
+                                    <hr/>
+                                    <p>First, enter your password to confirm you own this account</p>
+                                    <input name="password" type="password" required="1"
+                                        t-attf-class="form-control #{'is-invalid' if deactivate_error == 'password' else ''}"
+                                        placeholder="Password"/>
+                                    <div t-if="deactivate_error == 'password'" class="invalid-feedback">
+                                        Wrong password.
+                                    </div>
+                                    <a class="mt-3 d-block mb-2" href="/web/reset_password">Forgot password ?</a>
+                                    <hr/>
+                                    <p>
+                                        Second, confirm you want to deactivate your account by
+                                        copying down your login (<t t-esc="env.user.login"/>).
+                                    </p>
+                                    <input name="validation" type="text" required="1"
+                                        t-attf-class="form-control #{'is-invalid' if deactivate_error == 'validation' else ''}"/>
+                                    <div t-if="deactivate_error == 'validation'" class="invalid-feedback">
+                                        You should enter "<t t-esc="env.user.login"/>" to validate your action.
+                                    </div>
+                                    <div class="d-flex flex-row align-items-center">
+                                        <input type="checkbox" name="request_blacklist" id="request_blacklist" checked="1"/>
+                                        <label for="request_blacklist" class="ml-2 mw-100 font-weight-normal mt-3">
+                                            I do not wish to receive further marketing communications
```

Comment:
But people don't know our implementation uses a blacklist. If I see this option, I am understanding the opposite : "keep me in a list of people that we can contact later". I am not sure this option is very relevant. If we correctly remove all the info we have about somebody, he should no longer be contacted until he re-fills a contact form or something, right ?

Reply:
@mart-e It's not 100% sure, e.g. if you have an other partner with the same email

Reply:
I don't think that works in practice, users don't know/care about the tech details.

1. As mentioned, the message does not ask about being put into a block list, it asks about receiving further communication. How would they receive further communications if we deleted their data? It makes no sense for the end user, unless the deletion doesn't really do what they want. You see the problem?

2. The real answer to why we could have this option is quite technical: we may not be able to delete all copies of their data, or to delete it immediately, for legal reasons or due to technical complexities. That's quite complicated/confusing for users, we can't explain all that in the wizard.

I see 2 solutions:

A) Reword that option and make it "Put my email and phone in a block list to make sure I'm never contacted again". It's less confusing for end-users, but still surprising: I've never seen this option during an account deletion process.  (Note: please prefer "block list" for UX messages).
Now the risk is that some people will decide to skip it, will still receive marketing comm, and will then complain about it, because it's not the result they expected.

B) Do the block list implicitly for all users, with the small consequence that their email remains somewhere in the database, if there was really no other data left. Yet it will be in the *only place* where it won't be used for marketing. And that will better protect them in cases where there was indeed other data that couldn't be deleted. For all practical purposes, users will get the result they wanted.

Either option works for me, but not the current wording ;)

Reply:
@odony Ok to rename with `Put my email and phone in a block list to make sure I'm never contacted again` as the option is important to keep imo

Reply:
@mart-e The option cost almost nothing... Except a little sentence at the end of the modal...

Reply:
When I joined Odoo, we told me adding an extra option is rarely a good idea if we can find a defaut that suits most ;-)

Reply:
I guess it's just a question of POV, I think the option is important, and cost almost nothing from a UI / technical POV, because as a Odoo user, I would like to have the choice. ODO seems ok with the option as long as we reword it, LNA want the option as well, I don't think it's useful to write 500 lines of Github comment to debate about a checkbox ^^ (except if you think it's a very terrible idea)